cmd/pipeline-template/plan_test.go

```diff
@@ -0,0 +1,251 @@
+	err := rootCmd.Execute()
+	if err != nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+func TestPipelineTemplatePlan_stdin(t *testing.T) {
+	ts := gateServerPlanSuccess()
+	defer ts.Close()
+
+	tempFile := tempPipelineTemplateFile(testPlanConfig)
+	if tempFile == nil {
+		t.Fatal("Could not create temp pipeline template file.")
+	}
+	defer os.Remove(tempFile.Name())
+
+	// Prepare Stdin for test reading.
+	tempFile.Seek(0, 0)
+	oldStdin := os.Stdin
+	defer func() { os.Stdin = oldStdin }()
+	os.Stdin = tempFile
+
+	args := []string{"pipeline-template", "plan", "--gate-endpoint", ts.URL}
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err != nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+func TestPipelineTemplatePlan_fail(t *testing.T) {
+	ts := GateServerFail()
+	defer ts.Close()
+
+	tempFile := tempPipelineTemplateFile(testPlanConfig)
+	if tempFile == nil {
+		t.Fatal("Could not create temp pipeline template file.")
+	}
+	defer os.Remove(tempFile.Name())
+
+	args := []string{"pipeline-template", "plan", "--config", tempFile.Name(), "--gate-endpoint", ts.URL}
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Expected failure but command succeeded")
+	}
+}
+
+func TestPipelineTemplatePlan_flags(t *testing.T) {
+	ts := gateServerPlanSuccess()
+	defer ts.Close()
+
+	args := []string{"pipeline-template", "plan", "--gate-endpoint", ts.URL} // Missing pipeline config file and stdin.
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Expected failure but command succeeded")
+	}
+}
+
+// gateServerUpdateSuccess spins up a local http server that we will configure the GateClient
+// to direct requests to. Responds with 404 NotFound to indicate a pipeline template doesn't exist,
+// and Accepts POST calls.
+func gateServerPlanSuccess() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.Handle("/v2/pipelineTemplates/plan", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			fmt.Fprintln(w, strings.TrimSpace(testPipelineTemplatePlanResp))
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return httptest.NewServer(mux)
+}
+
+const testPipelineTemplatePlanResp = `
+{
+ "application": "app",
```

Comment:
Is this guaranteed to be alphabetical when rendered by spin?

Reply:
Yep.